web/packages/teleterm/src/ui/DocumentCluster/DocumentCluster.tsx

```diff
@@ -52,97 +53,144 @@ export default function DocumentCluster(props: {
+      )
+    )
+  );
 
   return (
     <Document visible={props.visible}>
-      <ClusterState
-        clusterUri={clusterUri}
-        rootCluster={rootCluster}
-        cluster={cluster}
-        onLogin={logIn}
-      />
+      <Layout>
+        <ClusterState
+          clusterName={clusterName}
+          clusterUri={clusterUri}
+          rootCluster={rootCluster}
+          cluster={cluster}
+          syncCluster={syncCluster}
+          clusterSyncAttempt={clusterSyncAttempt}
+        />
+      </Layout>
     </Document>
   );
 }
 
 function ClusterState(props: {
   clusterUri: uri.ClusterUri;
+  clusterName: string;
   rootCluster: Cluster;
   cluster: Cluster | undefined;
-  onLogin(): void;
+  syncCluster(): void;
+  clusterSyncAttempt: Attempt<void>;
 }) {
   if (!props.rootCluster.connected) {
     return (
-      <RequiresLogin clusterUri={props.clusterUri} onLogin={props.onLogin} />
+      <RequiresLogin
+        clusterName={props.clusterName}
+        syncCluster={props.syncCluster}
+        clusterSyncAttempt={props.clusterSyncAttempt}
+      />
     );
   }
 
   if (!props.cluster) {
-    return <NotFound clusterUri={props.clusterUri} />;
+    return <NotFound clusterName={props.clusterName} />;
   }
 
   if (props.cluster.leaf && !props.cluster.connected) {
-    return <LeafDisconnected clusterUri={props.clusterUri} />;
+    return (
+      <LeafDisconnected
+        clusterName={props.clusterName}
+        syncCluster={props.syncCluster}
+        clusterSyncAttempt={props.clusterSyncAttempt}
+      />
+    );
   }
 
+  return <UnifiedResources clusterUri={props.clusterUri} />;
+}
+
+function RequiresLogin(props: {
+  clusterName: string;
+  syncCluster(): void;
+  clusterSyncAttempt: Attempt<void>;
+}) {
   return (
-    <Layout>
-      <UnifiedResources clusterUri={props.clusterUri} />
-    </Layout>
+    <PrintState
+      clusterName={props.clusterName}
+      clusterState="Cluster is offline."
+      action={{
+        attempt: props.clusterSyncAttempt,
+        label: 'Connect',
+        run: props.syncCluster,
+      }}
+    />
   );
 }
 
-function RequiresLogin(props: { clusterUri: uri.ClusterUri; onLogin(): void }) {
+function LeafDisconnected(props: {
+  clusterName: string;
+  syncCluster(): void;
+  clusterSyncAttempt: Attempt<void>;
+}) {
   return (
-    <Flex
-      flexDirection="column"
-      mx="auto"
-      justifyContent="center"
-      alignItems="center"
-    >
-      <Text typography="h4" color="text.main" bold>
-        {props.clusterUri}
-        <Text as="span" typography="h5">
-          {` cluster is offline`}
-        </Text>
-      </Text>
-      <ButtonPrimary mt={4} width="100px" onClick={props.onLogin}>
-        Connect
-      </ButtonPrimary>
-    </Flex>
+    <PrintState
+      clusterName={props.clusterName}
+      clusterState="Trusted cluster is offline."
+      action={{
+        attempt: props.clusterSyncAttempt,
+        label: 'Reconnect',
```

Comment:
Maybe just "Refresh cluster status"? "Reconnect" makes it sound like the problem was that me, the user, did not connect the cluster and thus it's offline and I need to reconnect it.